framework/include/meshgenerators/SideSetsAroundSubdomainGenerator.h

```diff
@@ -28,24 +28,8 @@ class SideSetsAroundSubdomainGenerator : public SideSetsGeneratorBase
 
   std::unique_ptr<MeshBase> generate() override;
 
-protected:
-  std::unique_ptr<MeshBase> & _input;
-
-  /// names of the sidesets to which the faces will be added
-  std::vector<BoundaryName> _boundary_names;
-
-  /// true if only faces close to "normal" will be added
-  bool _using_normal;
-
-  /// Only apply the sideset to external boundaries
-  const bool _external_only;
   /**
-   * if normal is specified, then faces are only added
-   * if face_normal.normal_hat <= 1 - normal_tol
-   * where normal_hat = _normal/|_normal|
+   * Determine whether the given side of an element resides on an external or internal boundary
    */
-  Real _normal_tol;
-
-  /// if specified, then faces are only added if their normal is close to this
-  Point _normal;
+  static bool elemSideOnBoundary(const Elem * elem, const uint & side);
```

Comment:
why static?
```suggestion
  static bool elemSideOnBoundary(const Elem * const elem, const unsigned int side);
```

Reply:
I made it static because it does not rely on data from a particular `SideSetsAroundSubdomainGenerator` object. Would it make sense to instead move it out of the class into a utility function? I could also remove the static and make the method const, but I don't see why removing the static is necessary or desirable because the method doesn't need access to object data.

Reply:
`elemSideOnBoundary` is used twice in `generate`, but not anywhere else. Would it make sense to define it as a lambda function in `generate`?

Reply:
so is this routine more of a elemSideOnExternalBoundary ?
I dont see how we can check if it s on a particular boundary in the restricted set without using object data

Reply:
According to my docstring, the method determines "whether the given side of an element resides on an external or internal boundary"
Here is the implementation. It does not require the use of object data because the element and side in question are passed as arguments and the relevant neighbor element is extracted from those parameters.
https://github.com/idaholab/moose/blob/95a82d112c0259fe2ac54bab612c4abf6fb5ed7f/framework/src/meshgenerators/SideSetsAroundSubdomainGenerator.C#L186-L191

Reply:
We could also rename the method to `elemSideOnInternalOrExternalBoundary` if you think it would add clarity.

Reply:
We could also move this method into the base class.